wgpu-native/src/instance.rs

```diff
@@ -201,10 +201,23 @@ pub fn instance_get_adapter(instance_id: InstanceId, desc: &AdapterDescriptor) -
     };
 
     let (mut low, mut high, mut other) = (None, None, None);
+    // On Windows > 1803, dx12 enumerate_adapters returns the adapters in order from highest to
+    // lowest performance. Therefore, the first found adapter in each category is selected.
+    //
+    // TODO: move power/performance policy querying into gfx, which has more context into
+    // performance policy than wgpu
     for adapter in adapters {
         match adapter.info.device_type {
-            hal::adapter::DeviceType::IntegratedGpu => low = Some(adapter),
-            hal::adapter::DeviceType::DiscreteGpu => high = Some(adapter),
+            hal::adapter::DeviceType::IntegratedGpu => low = low.or(Some(adapter)),
+            hal::adapter::DeviceType::DiscreteGpu => {
+                high = match desc.power_preference {
+                    // If `LowPower`, prefer lowest power `DiscreteGPU`
```

Comment:
it would be good to only have this match once, and we have it afterwards
Can we refactor this code to record `high_first` and `high_last` instead of having a match here?

Reply:
Done!